test/suites/smoke-test-dancelight/test-authority-consistency-assignation.ts

```diff
@@ -17,22 +17,32 @@ describeSuite({
             id: "C01",
             title: "Collator assignation and authority assignation should match with observed mapping in containers",
             test: async () => {
-                const assignmentCollatorAccount = (
-                    await api.query.tanssiCollatorAssignment.collatorContainerChain()
-                ).toJSON();
+                const assignmentCollatorAccount = await api.query.tanssiCollatorAssignment.collatorContainerChain();
                 const sessionIndex = (await api.query.session.currentIndex()).toNumber();
                 const assignmentCollatorKey = (
                     await api.query.tanssiAuthorityAssignment.collatorContainerChain(sessionIndex)
-                ).toJSON();
```

Comment:
We abuse toJSON because it makes it easier to see the diff when an expect fails. Can you verify that expect(assignmentCollatorKey).toBeNull() shows a nice error message with the assignmentCollatorKey?

Reply:
Will check. Problem about using toJSON() is that it casts it to `any`. What we currently have is that it will show you an error without even having to run the tests

Reply:
@timbrinded please check this expect thing

Reply:
When it fails it'll still show that blah does not contain "5DFi8fa9gABiFpWsSosS5xVBwGVy28crWmh1J…"

The added benefit here is that if it cant find it due to test error (i.e. the shape of the struct changes) then it's more obvious then what we get right now which is "cannot find .toHex() on undefined"

Reply:
When I do this:

```
expect(assignmentCollatorKey).toBe(null);
```

I see this, with about 10000 lines of type stuff:

![imagen](https://github.com/user-attachments/assets/6de493ac-8df8-4a5b-8f57-fdf95cb2e0e0)

... many lines more in between ....

![imagen](https://github.com/user-attachments/assets/06f36b3c-6142-4fff-8341-8bfc4dd915c6)



On the other hand, if I use `.toJSON()`:

```
expect(assignmentCollatorKey.toJSON()).toBe(null);
```

I see this:

![imagen](https://github.com/user-attachments/assets/e3291361-d414-4edc-a401-769a964677ab)

Much nicer

Reply:
Same applies to console log, removing toJSON makes debugging harder

Reply:
I agree that toJSON is great for printing to console (and you can still do that!)

It makes for brittle and opaque test cases though, which in my experience with moonbeam you want to optimize to remove. Test case maintainance is a pain because you end up having to debug features you worked on years ago and forgot how it work. In that scenario a magic blob of JSON is really unhelpful because you dont know what the correct shape is.

Using the proper PolkadotJS types means you can do things like:
`expect(assignmentCollatorKey.isEmpty).toBe(true);`

to replicate your example.

Reply:
Furthermore when you change the shape of the underlying structures you can pick up those errors at compile time instead of run time

Reply:
@tmpolaczyk I've logged already on most test cases:
![image](https://github.com/user-attachments/assets/8dae15ae-5e2b-4f7c-82b0-ea02b25d5866)

But i'll go through them all now and see if theres any other places you'd want to see it.

Reply:
ok we should have best of both worlds now:
![image](https://github.com/user-attachments/assets/b6c05034-0478-4416-b84e-925a198129b5)